interfaces.go

```diff
@@ -0,0 +1,21 @@
+package vfsgen
+
+// GzipByter is implemented by compressed files for
+// efficient direct access to the internal compressed bytes.
+type GzipByter interface {
+	// GzipBytes returns gzip compressed contents of the file.
+	GzipBytes() []byte
+}
+
+// TODO: Choose one from below.
+
+// NotWorthGzipCompressing is implemented by files that were determined
+// not to be worth gzip compressing (the file size did not decrease as a result).
+type NotWorthGzipCompressing interface {
+	NotWorthGzipCompressing()
+}
+
+// NotWorthGzipCompressing indicates the file is not worth gzip compressing.
+type NotWorthGzipCompressing_ interface {
```

Comment:
> A suffixed `_` seems strange to me.

See the `// TODO: Choose one from below.` comment above.

> If you don't mind me asking, what is the value in knowing that a file is "not worth gzip compressing" explicitly instead of just having it NOT implement the `GzipByter` interface?

Absolutely.

An http server like `http.FileServer` that additionally does gzip-compression can check if this interface is implemented and skip gzip compression for that file.

If the `GzipByter` interface is not implemented, it might be because the file is not worth gzip compressing, or because `vfsgen` was not used to generate the `http.File`. For example, if you're using `http.Dir`.

Maybe seeing this code will be helpful in making the above logic more clear:

https://github.com/shurcooL/go/blob/5b4bf3db71a69c22472000aa60576680c1d10888/gzip_file_server/gzip_file_server.go#L81-L114